adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/MqttContext.java

```diff
@@ -270,6 +282,8 @@ public String tenant() {
 
         if (authenticatedDevice != null) {
             return authenticatedDevice.getTenantId();
+        } else if (mappedTargetAddress.isPresent() && !Strings.isNullOrEmpty(mappedTargetAddress.get().getTenantId())) {
+            return mappedTargetAddress.get().getTenantId();
```

Comment:
I now think that we can omit this change here - the mapped target address isn't supposed to contain a different tenant id anyway.

Reply:
FMPOV we actually need to enforce that a mapper can not change the tenant ID

Reply:
Do you want me to update this and add a specific check for this? Implicitly this is not possible now, since the mappedTargetAddress is not used to retrieve the tenant from

Reply:
I agree. For Hono 1.12.0 I think we could simply just use `MappedMessage#targetDeviceId` instead of `MappedMessage#targetAddress`.
I think this is out of scope for this PR though.
(At least with the current only `MessageMapping` implementation (`HttpBasedMessageMapping`), the tenant id cannot be overridden now anyway.)